xarray/core/dataarray.py

```diff
@@ -1194,13 +1194,22 @@ def to_dict(self):
         Convert this xarray.DataArray into a dictionary following xarray
         naming conventions.
 
-        Useful for coverting to json.
+        Useful for coverting to json.To avoid datetime incompatibility
+        use decode_times=False kwarg in xarrray.open_dataset.
 
         See also
         --------
         xarray.DataArray.from_dict
         """
-        d = {'coords': {}, 'attrs': dict(self.attrs), 'dims': self.dims}
+        def np_check(attrs):
+            # catch numpy arrays in attrs and convert to nested lists
+            attrs = dict(attrs)
+            for k, v in attrs.items():
+                if hasattr(v, 'tolist'):
```

Comment:
Let's also fix NumPy scalars values, e.g., `np.float32(0)`. To convert these, use `.item()`.